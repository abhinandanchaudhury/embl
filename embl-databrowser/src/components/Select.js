import React, { Component } from 'react';

class Select extends Component {
    getOptionTag(option, i) {
        return (
          <option value={option.name} key={i}>{option.display_name} [ {option.name} ]</option>
        );
    }

    getOptionTags(options) {
        return options.map((option, i) => {
          return this.getOptionTag(option, i);
        });
    }

    getOptions() {
        var options = this.props.options;
        if (this.props.grouping) {
          return this.getOptgroupTags(options);
        } else {
          return this.getOptionTags(options);
        }
    }
    getOptgroupTags(groups) {
        var optgroups = groups.map((group, t) => {
          var children = this.getOptionTags(group[this.props.grouping]);
          return (
            <optgroup key={t} label={group.group}>
              {children}
            </optgroup>
          );
    });
    return optgroups;
    }

    render() {
        var options = this.getOptions();
        return (
          <select className="form-control" defaultValue={this.props.selected} {...this.props} >
            <option>{this.props.placeholder}</option>
            {options}
          </select>
        );
    }
}
Select.defaultProps = {
      options: [],
      grouping: null,
      selected: null
}

export default Select;