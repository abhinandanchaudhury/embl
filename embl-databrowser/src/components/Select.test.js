import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, configure } from 'enzyme';
import Select from './Select';
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() })
describe("Test the Select Component", () => {
    it('Select renders without crashing', () => {
      const species = [{name:'homo_sapiens', display_name: 'Human'}];
      let selectedSpecies = 'homo_sapiens';
      const wrapper = shallow(<Select id="species" options={species} placeholder={'Please select the species'} selected={selectedSpecies} onChange={() => {}} />);
      expect(wrapper.prop('options')).toBe(species);
    });
    it('Select renders with properties', () => {
          const species = [{name:'homo_sapiens', display_name: 'Human'}];
          let selectedSpecies = 'homo_sapiens';
          const wrapper = shallow(<Select id="species" options={species} placeholder={'Please select the species'} selected={selectedSpecies} onChange={() => {}} />);
          expect(wrapper.prop('placeholder')).toBe('Please select the species');
    });
    it('Testing option values', () => {
              const species = [{name:'homo_sapiens', display_name: 'Human'}];
              let selectedSpecies = 'homo_sapiens';
              const wrapper = shallow(<Select id="species" options={species} placeholder={'Please select the species'} selected={selectedSpecies} onChange={() => {}} />);
              const instance = wrapper.instance();
              instance.getOptions();
              expect(wrapper.find("select").props("children").defaultValue).toBe(species[0].name);
              expect(wrapper.find("select").props("children").selected).toBe(species[0].name);
              expect(wrapper.find("select").props("children").options[0].display_name).toBe(species[0].display_name);
    });
    it('Snapshot testing', () =>{
        const species = [{name:'homo_sapiens', display_name: 'Human'}];
        const selectedSpecies = 'homo_sapiens';
        const tree = renderer.create(
          <Select id="species" options={species} placeholder={'Please select the species'} selected={selectedSpecies} onChange={() => {}} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
