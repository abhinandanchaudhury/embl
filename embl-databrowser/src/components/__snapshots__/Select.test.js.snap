// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Select Component Snapshot testing 1`] = `
<select
  className="form-control"
  defaultValue="homo_sapiens"
  grouping={null}
  id="species"
  onChange={[Function]}
  options={
    Array [
      Object {
        "display_name": "Human",
        "name": "homo_sapiens",
      },
    ]
  }
  placeholder="Please select the species"
  selected="homo_sapiens"
>
  <option>
    Please select the species
  </option>
  <option
    value="homo_sapiens"
  >
    Human
     [ 
    homo_sapiens
     ]
  </option>
</select>
`;
